### PR TITLE
RHPAM-1029: Smart Router Servers view error

### PR DESCRIPTION
**Pull request by karreiro**
See: https://issues.jboss.org/browse/RHPAM-1029

Before:
_before

After:
_after

### CODE REVIEW

**Comment by karreiro** [comment]
Jenkins please retest this.

**Comment by karreiro** [comment]
@mswiderski I'm still checking the broken tests. But, I would really appreciate your feedback.

Thanks.

File: kie-server-parent/kie-server-controller/kie-server-controller-impl/src/test/java/org/kie/server/controller/impl/KieServerInstanceManagerTest.java [hunk 1]
```diff
@@ -470,6 +483,58 @@ public void testUpgradeAndStartContainer() {
...
+        final ServerInstanceKey serverInstanceKey = mock(ServerInstanceKey.class);
+        final Collection<ServerInstanceKey> serverInstanceKeys = Collections.singletonList(serverInstanceKey);
+
+        when(serverTemplate.getServerInstanceKeys()).thenReturn(serverInstanceKeys);
+
+        doReturn(client).when(instanceManager).getClient(any());
+
+        final List<Container> containers = instanceManager.callRemoteKieServerOperation(serverTemplate, containerSpec, operation);
+
+        assertFalse(containers.isEmpty());
```
Comment (jomarko) [inline kie-server-parent/kie-server-controller/kie-server-controller-impl/src/test/java/org/kie/server/controller/impl/KieServerInstanceManagerTest.java]: I think we should assert precise size of the `containers` collection.

File: kie-server-parent/kie-server-router/kie-server-router-proxy/src/test/java/org/kie/server/router/KieServerRouterTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    public void setup() {
+
+        when(blockingHttpExchange.getSender()).thenReturn(sender);
+        when(serverConnection.getByteBufferPool()).thenReturn(bufferPool);
+
+        kieServerRouter = new KieServerRouter();
+    }
+
+    @Test
+    public void testMakePathHandler() throws Exception {
```
Comment (jomarko) [inline kie-server-parent/kie-server-router/kie-server-router-proxy/src/test/java/org/kie/server/router/KieServerRouterTest.java]: We miss negative case test method:
```java
@Test
public void testMakePathHandler_Invalid() throws Exception {
    assertInvalidPath("/@#$#");
    // ...
    // ...
}
```

File: kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/KieServerRouter.java [hunk 3]
```diff
@@ -141,6 +129,24 @@ public void start(String host, Integer port, ConfigurationListener ...listeners)
 
     }
 
+    PathHandler makePathHandler(final AdminHttpHandler adminHandler,
```
Comment (mswiderski) [inline kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/KieServerRouter.java]: I guess this is more of refactoring rather fix for the issue, right?

**Comment by karreiro** [comment]
Thanks for the reviews, everyone.

@mswiderski @jhrcek When I was checking the broken tests, I noticed that my previous changes had some impact in other APIs. Thus, I've changed my approach to avoid regressions.

I could fix the router API and get the same result by changing the `DefaultRestrictionPolicy.java` class:
_demo

@jomarko Changes applied.

Thanks.

**Comment by karreiro** [comment]
Unrelated test failures.

**Pull request merged**